core/test/forest/LocalLinearForestTest.cpp

```diff
@@ -105,3 +105,29 @@ TEST_CASE("LLF predictions vary linearly with Y", "[local_linear, forest]") {
   REQUIRE(equal_doubles(delta / predictions.size(), 1, 1e-1));
   delete data;
 }
+
+TEST_CASE("LLCF predictions are reasonable", "[local_linear], [causal], [forest]") {
```

Comment:
Is this meant as a characterization test (i.e., you're just checking that predictions won't change as you do refactors/ make performance improvements)? If so, it would be great to write it as a C++ test with the other cases in `ForestCharacterizationTest.cpp`.

Or maybe you're trying to sanity check the output of the forest? In that case, we tend to prefer invariant checks, since from C++ it can be hard to debug what's happening statistically. The suite `RegressionForestTest` has some examples of checking invariants, like 'honest regression forests are shift invariant'.